proto/redpanda/api/dataplane/v1/quota.proto

```diff
@@ -0,0 +1,222 @@
+    (buf.validate.field).enum = {
+      defined_only: true
+      not_in: [
+        0,
+        1,
+        4,
+        5
+      ] /* Disallow unspecified, any, user, and ip */
+    },
+    (buf.validate.field).required = true
+  ];
+
+  // Entity can be defined by its name or use the default value.
+  enum EntityRequestType {
+    ENTITY_REQUEST_TYPE_UNSPECIFIED = 0;
+    ENTITY_REQUEST_TYPE_NAME = 1;
+    ENTITY_REQUEST_TYPE_DEFAULT = 2;
+  }
+
+  EntityRequestType entity_request_type = 2 [(buf.validate.field).required = true];
+
+  // Either the client ID or the client ID prefix, depending on the entity_request_type
+  // If entity_request_type is ENTITY_REQUEST_TYPE_DEFAULT, this field can be empty
+  string entity_name = 3;
+}
+
+message CreateQuotaRequest {
+  message RequestValue {
+    Quota.ValueType value_type = 1 [
+      (buf.validate.field).enum = {
+        defined_only: true
+        not_in: [
+          0,
+          1
+        ] /* Disallow unspecified and any */
+      },
+      (buf.validate.field).required = true
+    ];
+    int64 value = 2;
+  }
+
+  // To specify multiple entities to apply the same quota
+  repeated RequestEntity entities = 1 [(buf.validate.field).required = true];
+  repeated RequestValue values = 2 [(buf.validate.field).required = true];
+}
+
+message CreateQuotaResponse {}
+
+message DeleteQuotaRequest {
+  option (buf.validate.message).cel = {
+    id: "value_type_not_unspecified_or_any"
+    message: "value_type cannot be VALUE_TYPE_UNSPECIFIED (0) or VALUE_TYPE_ANY (1)"
+    expression: "this.value_type != 0 && this.value_type != 1"
+  };
+
+  RequestEntity entity = 1 [(buf.validate.field).required = true];
+
+  // List of values that should be deleted for the specified entities
+  Quota.ValueType value_type = 2 [(buf.validate.field).required = true];
+}
+
+message DeleteQuotaResponse {}
+
+service QuotaService {
+  option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_tag) = {
+    name: "Quotas"
+    description: "Manage client throughput limits. See [Redpanda Client Throughput Limits](https://docs.redpanda.com/current/manage/cluster-maintenance/manage-throughput/#client-throughput-limits) for more information."
+  };
+
+  rpc ListQuotas(ListQuotasRequest) returns (ListQuotasResponse) {
+    option (google.api.http) = {get: "/v1/quotas"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List quotas"
+      description: "List client quotas. Use filter parameters to narrow results by entity type and filter criteria."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.ListQuotasResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_KAFKA
+    };
+  }
+
+  rpc CreateQuota(CreateQuotaRequest) returns (CreateQuotaResponse) {
```

Comment:
NIT: I would rename to SetQuotas since it's idempotent and a bit more of an upsert than create under the hood?